src/language/api/compilationStatus.ts

```diff
@@ -13,6 +14,7 @@ export interface ICompilationStatusParams {
   uri: DocumentUri;
   version?: integer;
   status: CompilationStatus;
+  message?: string; // FIXME this expects a string or undefined but it receives null
```

Comment:
Could you resolve the FIXME?

Reply:
Yep, I need to figure how to do that ^^

Reply:
Maybe @camrein can advise? 

Hi @camrein, if you have time: I have a problem with the the type of the `message?` field.  On the C# side the types is `string?`, and when serializing a `message` with value `null` the C# server produces `"message": null` in JSON (instead of leaving it undefined).  I've changed to `message?: string | null` on the JS side, which if fine but a bit ugly.  Can you think of something better?

Relatedly: I see that we have `version?: int` above; are we sure that the server omits the `version` field when `version` is `null`, or should that have been `version?: int | null` on the extension side?

Reply:
> Hi @camrein, if you have time: I have a problem with the the type of the `message?` field.

The problem, as you might already think, is that C# doesn't know `undefined`.
To correctly represent this, you'd have to express this with a type within the language server. For example, create a sub-type of `CompilationStatusParams` such as `CompilationStatusWithMessageParams`.

> Relatedly: I see that we have `version?: int` above; are we sure that the server omits the `version` field when `version` is `null`, or should that have been `version?: int | null` on the extension side?

Not really. It'll be sent as `null` as well; thus, the correct type would be `version: int | null`. I copied this style from OmniSharp. However, this style is primarily used for client-to-server communication so it's not entirely correct. We could omit the `version` field since it's not used yet.

Side-note: As of now, apart from one or two exceptions, the code within the extension treats `undefined` and `null` the same.

Reply:
Thanks for the clarification.  Given this it's probably fine to stick with what I did (make the type of `message?` be `string | null`)!